### PR TITLE
Unexpected behavior note

### PR DESCRIPTION
**Pull request by alflennik**
Makes the unexpected behavior note available for all unexpected behaviors instead of just "Other."

Goes along with a PR in the ARIA-AT repo which is here: https://github.com/w3c/aria-at/pull/802 The two PRs can be merged in any order and at different times if needed. But following review changes to one might require changes to the other.

### CODE REVIEW

File: server/resolvers/TestResultOperations/saveTestResultCommon.js [hunk 1]
```diff
@@ -95,23 +94,31 @@ const assertTestResultIsValid = newTestResult => {
...
-            (!!unexpectedBehavior.otherUnexpectedBehaviorText &&
-                unexpectedBehavior.id !== 'OTHER') ||
-            (!unexpectedBehavior.otherUnexpectedBehaviorText &&
-                unexpectedBehavior.id === 'OTHER')
-        ) {
+    const checkUnexpectedBehavior = (
+        unexpectedBehavior,
+        unexpectedBehaviorNote
+    ) => {
+        if (!unexpectedBehaviorNote && unexpectedBehavior.id === 'OTHER') {
```
Comment (evmiguel) [inline server/resolvers/TestResultOperations/saveTestResultCommon.js:101]: Non-blocking: Are we still expecting a behavior note to be filled when OTHER? I thought the note would be decoupled from all undesirable behaviors.

File: client/resources/aria-at-test-io-format.mjs [hunk 2]
```diff
@@ -460,8 +473,8 @@ class UnexpectedInput {
   static fromBuiltin() {
     return new UnexpectedInput({
       behaviors: [
-        ...UNEXPECTED_BEHAVIORS.map(description => ({description, requireExplanation: false})),
-        {description: "Other", requireExplanation: true},
+        ...UNEXPECTED_BEHAVIORS.map(description => ({ description, requireExplanation: false })),
+        { description: 'Other', requireExplanation: true },
```
Comment (evmiguel) [inline client/resources/aria-at-test-io-format.mjs:477]: Same question here. Is Other supposed to be coupled to the explanation?

File: client/resources/aria-at-harness.mjs [hunk 3]
```diff
@@ -371,47 +393,69 @@ function renderVirtualInstructionDocument(doc) {
...
-                  )
+            label(
+              forInput(`${failOption.description}-${commandIndex}`),
+              rich(failOption.description)
+            ),
+            br()
+          )
+        ),
+        fragment(
+          // TODO: Figure out why this isn't appearing
```
Comment (evmiguel) [inline client/resources/aria-at-harness.mjs:443]: Looks like this is appearing!

**Review by evmiguel — CHANGES_REQUESTED** [review]
Looking good! There's a bug in the form validation. 

https://user-images.githubusercontent.com/2789098/184236046-840ae019-fada-4e5f-be99-bd0b0fc83652.mov

UPDATE: Actually, the bug persists when switching between No and Yes.

File: client/resources/aria-at-test-run.mjs [hunk 4]
```diff
@@ -207,39 +209,41 @@ export function instructionDocument(resultState, hooks) {
           {
             required: true,
             highlightRequired: resultStateCommand.atOutput.highlightRequired,
-            description: "(required)",
+            description: '(required)',
```
Comment (evmiguel) [inline client/resources/aria-at-test-run.mjs:212]: I'm not seeing this. How do I get to this state?

File: client/resources/aria-at-test-run.mjs [hunk 5]
```diff
@@ -263,54 +268,73 @@ export function instructionDocument(resultState, hooks) {
...
+              {
+                required: resultUnexpectedBehavior.behaviors.some(
+                  ({ checked, requireExplanation }) => requireExplanation && checked
+                ),
+                highlightRequired:
+                  resultState.currentUserAction === 'validateResults' &&
+                  resultUnexpectedBehavior.behaviors.some(
+                    ({ checked, requireExplanation }) => requireExplanation && checked
+                  ),
+                description: resultUnexpectedBehavior.behaviors.some(
```
Comment (evmiguel) [inline client/resources/aria-at-test-run.mjs:319]: I'm not seeing this. How do I get to this state?

**Comment by alflennik** [comment]
Superseded by https://github.com/w3c/aria-at-app/pull/596

**Pull request closed without merging**